File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int ans = 0, n = nums.size();
        unordered_map<int, int>mp;
        mp[0] = 1;
        for(int i=0, sum = 0; i<n; i++) {
            sum += nums[i];
            if(mp.find(sum - k) != mp.end()) {
                ans += mp[sum - k];
            }
            mp[sum]++;
        }
        return ans;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int ans = 0, n = nums.length, sum = 0;
        Map<Integer, Integer>mp = new HashMap<>();
        mp.put(sum, 1);
        for(int i=0; i<n; i++) {
            sum += nums[i];
            if(mp.containsKey(sum - k)) {
                ans += mp.get(sum - k);
            }
            mp.put(sum, mp.getOrDefault(sum, 0) + 1);
        }
        return ans;
    }
}